src/pages/dogs/[id].jsx

```diff
@@ -30,12 +38,47 @@ export default function IndividualDogPage() {
     }
   }, [router.query]);
 
-  if (!data || data === undefined || !data.success) {
+  useEffect(() => {
+    let logSearch = {};
+    if (logFilters) {
+      Object.keys(logFilters)
+        .filter(
+          (category) =>
+            category && Object.values(logFilters[category]).length > 0
+        )
+        .forEach((category) => {
+          logSearch[category] = Object.values(logFilters[category]);
+        });
+    }
+
+    logSearch.title = logSearchFilter;
+
+    fetch("/api/logs/search", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(logSearch),
+    })
+      .catch((err) => setLogData([]))
+      .then((res) => res.json())
+      .then((logData) => setLogData(logData));
+  }, [logSearchFilter, logFilters]);
+
+  if (
+    !data ||
+    data === undefined ||
```

Comment:
this may be redundant. I'm pretty sure `undefined` is a falsey value in js. Double check this to make sure though